main.js

```diff
@@ -0,0 +1,207 @@
+import {
+  getProductList,
+  getProduct,
+  createProduct,
+  patchProduct,
+  deleteProduct,
+} from "./ProductService.js";
+
+import {
+  getArticleList,
+  getArticle,
+  createArticle,
+  patchArticle,
+  deleteArticle,
+} from "./ArticleService.js";
+
+/* ======================= Product 클래스 생성 ======================= */
+
+class Product {
+  constructor(name, description, price, tags, images, favoriteCount = 0) {
+    this.name = name;
+    this.description = description;
+    this.price = price;
+    this.tags = tags;
+    this.images = images;
+    this._favoriteCount = favoriteCount;
+  }
+
+  favorite() {
+    this._favoriteCount += 1;
+  }
+
+  get favoriteCount() {
+    return this._favoriteCount;
+  }
+
+  set favoriteCount(value) {
+    throw new Error("좋아요는 수정할 수 없습니다.");
+  }
+}
+
+// const product1 = new Product(
+//   "책상",
+//   "높낮이 조절가능",
+//   300000,
+//   "가구",
+//   "https://deskgood.com"
+// );
+// product1.favorite();
+// console.log(product1);
+
+/* =================== ElectronicProduct 클래스 생성 =================== */
+
+class ElectronicProduct extends Product {
+  constructor(
+    name,
+    description,
+    price,
+    tags,
+    images,
+    manufacturer,
+    favoriteCount
+  ) {
+    super(name, description, price, tags, images, favoriteCount);
+    this.manufacturer = manufacturer;
+  }
+}
+
+// const electronicProduct1 = new ElectronicProduct(
+//   "모니터",
+//   "QHD 4k",
+//   "400000",
+//   "전자제품",
+//   "https://monitor.co.kr",
+//   "LG"
+// );
+// electronicProduct1.favorite();
+// console.log(electronicProduct1);
+
+/* ======================= Article 클래스 생성 ======================= */
+
+class Article {
+  constructor(title, content, writer, likeCount = 0, createdAt = new Date()) {
+    this.title = title;
+    this.content = content;
+    this.writer = writer;
+    this._likeCount = likeCount;
+    this.createdAt = createdAt;
+  }
```

Comment:
```suggestion
  constructor(title, content, writer, likeCount = 0)) {
    this.title = title;
    this.content = content;
    this.writer = writer;
    this._likeCount = likeCount;
    this.createdAt =  new Date();
  }
```
createdAt도 주입을 받기 보다는 생성자에서 선언하여 생성자단에서 컨트롤 할 수 있게 해주면 좋을 것 같습니다.